### PR TITLE
Fix search when clicking on a category with slash in Smart Suggest

### PR DESCRIPTION
**Pull request by fin-markus**
Hi Isi,

this will fix the issue when clicking on a category in the Smart Suggest which has a slash in the name. Problem is on SW side because the `Router()->assemble()`-method can't handle the slashes in the url.

Best wishes, Markus

### CODE REVIEW

File: FinSearchAPI/FinSearchAPI.php [hunk 1]
```diff
@@ -507,27 +507,41 @@ public function before_searchIndexAction(\Enlight_Hook_HookArgs $args)
...
+                        $params = array();
+                        $params['sSearch'] = $request->getParam('sSearch');
+                        unset($requestedParams['sSearch']);
+
+                        /**
+                         * Set catFilter as additional parameter as it could break search when there
+                         * is a slash / in the category. Shopware can not handle that when using
+                         * Router()->assemble()
+                         */
+                        if ($requestedParams['catFilter'] && $requestedParams['catFilter'] !== "") {
```
Comment (ihanli) [inline FinSearchAPI/FinSearchAPI.php]: `!empty` should do the same

File: FinSearchAPI/FinSearchAPI.php [hunk 2]
```diff
@@ -507,27 +507,41 @@ public function before_searchIndexAction(\Enlight_Hook_HookArgs $args)
...
+                            $params['catFilter'] = $requestedParams['catFilter'];
+                            unset($requestedParams['catFilter']);
+                        }
+
+                        if ($requestedParams['attrib']) {
+                            foreach ($requestedParams['attrib'] as $filter => $filterValues) {
+                                if (is_array($filterValues)) {
+                                    foreach ($filterValues as $key => $filterValue) {
+                                        if ($filter == 'cat') {
```
Comment (ihanli) [inline FinSearchAPI/FinSearchAPI.php]: Not really an issue but still `==` -> `===`

File: FinSearchAPI/FinSearchAPI.php [hunk 3]
```diff
@@ -507,27 +507,41 @@ public function before_searchIndexAction(\Enlight_Hook_HookArgs $args)
...
+                         * is a slash / in the category. Shopware can not handle that when using
+                         * Router()->assemble()
+                         */
+                        if ($requestedParams['catFilter'] && $requestedParams['catFilter'] !== "") {
+                            $params['catFilter'] = $requestedParams['catFilter'];
+                            unset($requestedParams['catFilter']);
+                        }
+
+                        if ($requestedParams['attrib']) {
+                            foreach ($requestedParams['attrib'] as $filter => $filterValues) {
```
Comment (ihanli) [inline FinSearchAPI/FinSearchAPI.php]: Check if there is a PHP function which could already do that. If there isn't, looks like a piece of code that should go into a helper?

Comment (ihanli) [reply FinSearchAPI/FinSearchAPI.php]: Check array_map and array_walk. Those might reduce the lines of code.

File: FinSearchAPI/FinSearchAPI.php (hunk 1 above)
Comment (mmachatschek) [reply FinSearchAPI/FinSearchAPI.php]: Done

File: FinSearchAPI/FinSearchAPI.php (hunk 2 above)
Comment (mmachatschek) [reply FinSearchAPI/FinSearchAPI.php]: Done

**Pull request merged**